client/checkout/api/index.js

```diff
@@ -207,9 +205,9 @@ export default class WCPayAPI {
 							result.error.setup_intent.id );
 
 					// In case this is being called via payment request button from a product page,
```

Comment:
>In case this is being called via payment request button from a product page

If this line is meant for cases where the Payment Request Button is called from a product page, it might no longer be relevant since you’re removing the PRB implementation, right?

Additionally, to align with the comment, shouldn’t the code try to use `getConfig` first, like this?

```js
const ajaxUrl = getConfig( 'ajaxUrl' ) ?? getExpressCheckoutConfig( 'ajax_url' );
```

Reply:
Those are good questions, and I honestly don't know the answers to them :D 
I couldn't find anything in the PRB implementation that would trigger these functions, so I'm not sure when or the reasons why this was introduced.
The comment says:
> the getConfig function won't work

So it would make sense to have `getPaymentRequestData` be called first.

It could be that the implementation over the years changed so much that this would have no longer been relevant anymore in either case.
When a payment is being attempted, the backend could return a `redirect_url` with a #wcpay-confirm-....` hash prefix, used to confirm the PI.
In turn, the JS will call this function.
The backend has the ability to return such URL for PRBs, ECE, and any other payment method.
So I'd be wary of removing it or changing it, since the `confirmIntent()` function could be called from the ECE as well.